debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/macho/MachoDumpReader.java

```diff
@@ -355,27 +362,32 @@ public long getProcessId()
 		return 0;
 	}
 
-	public List<? extends IOSThread> getThreads()
+	public List<? extends IOSThread> getThreads() throws CorruptDataException
 	{
-		List<ThreadState> x86ThreadStates = new ArrayList<>(dumpFile.threads.states.size());
-		for(ThreadState state : dumpFile.threads.states) {
-			if (state.flavor == ThreadCommand.x86_THREAD_STATE) {
-				x86ThreadStates.add(state);
-			} else if (_signalNumber == -1 && state.flavor == ThreadCommand.x86_EXCEPTION_STATE) {
-				_signalNumber = state.registers.get("err").intValue();
+		List<IOSThread> threads = new ArrayList<>(dumpFile.threads.size());
+		for (ThreadCommand command : dumpFile.threads) {
+				OSXThread nativeThread = new OSXThread(0, command);
+				threads.add(nativeThread);
+				if (_signalNumber == -1) {
+					// will use the first thread's exception values, which should be the primary thread
+					try {
+						_signalNumber = nativeThread.registers.get("err").intValue();
+					} catch (NullPointerException e) {
```

Comment:
Catching `NullPointerException` is almost never the best way to deal with `null`. I expect this only needs to put `nativeThread.registers.get("err")` in a local and check that it's not null.

Reply:
@keithc-ca changed the code to actually check if the key exists